homeassistant/components/anthropic/config_flow.py

```diff
@@ -128,21 +132,48 @@ async def async_step_init(
     ) -> ConfigFlowResult:
         """Manage the options."""
         options: dict[str, Any] | MappingProxyType[str, Any] = self.config_entry.options
+        errors: dict[str, str] = {}
 
         if user_input is not None:
             if user_input[CONF_RECOMMENDED] == self.last_rendered_recommended:
                 if user_input[CONF_LLM_HASS_API] == "none":
                     user_input.pop(CONF_LLM_HASS_API)
-                return self.async_create_entry(title="", data=user_input)
 
-            # Re-render the options again, now with the recommended options shown/hidden
-            self.last_rendered_recommended = user_input[CONF_RECOMMENDED]
+                thinking_budget = user_input.get(
+                    CONF_THINKING_BUDGET, RECOMMENDED_THINKING_BUDGET
+                )
+
+                extended_thinking_enabled: bool = thinking_budget >= MIN_THINKING_BUDGET
+
+                if (
+                    extended_thinking_enabled
+                    and user_input.get(CONF_CHAT_MODEL, RECOMMENDED_CHAT_MODEL)
+                    not in THINKING_MODELS
+                ):
+                    errors[CONF_THINKING_BUDGET] = "model_does_not_support_thinking"
+
+                if (
+                    extended_thinking_enabled
+                    and user_input.get(CONF_TEMPERATURE, RECOMMENDED_TEMPERATURE) != 1.0
+                ):
+                    errors[CONF_TEMPERATURE] = "thinking_budget_incompatible"
+
+                if thinking_budget >= user_input.get(
+                    CONF_MAX_TOKENS, RECOMMENDED_MAX_TOKENS
+                ):
+                    errors[CONF_MAX_TOKENS] = "thinking_budget_too_large"
```

Comment:
Can we inject the max tokens in the placeholder to let user know what the max is?

Reply:
I've moved the error to the `thinking_budget` section, this way the user would just need to decrease the budget, not increase the max tokens to an unknown value.